packages/brick_offline_first_with_supabase/lib/src/offline_first_with_supabase_repository.dart

```diff
@@ -137,6 +147,220 @@ abstract class OfflineFirstWithSupabaseRepository
     await offlineRequestQueue.client.requestManager.migrate();
   }
 
+  @override
+  Future<void> notifySubscriptionsWithLocalData<TModel extends OfflineFirstWithSupabaseModel>({
+    bool notifyWhenEmpty = true,
+    Map<Query?, StreamController<List<OfflineFirstWithSupabaseModel>>>? subscriptionsByQuery,
+  }) async {
+    final supabaseControllers = supabaseRealtimeSubscriptions[TModel]
+        ?.values
+        .fold(<Query, StreamController<List<OfflineFirstWithSupabaseModel>>>{}, (acc, eventMap) {
+      acc.addEntries(eventMap.entries);
+      return acc;
+    });
+    await super.notifySubscriptionsWithLocalData<TModel>(
+      notifyWhenEmpty: notifyWhenEmpty,
+      subscriptionsByQuery: {
+        ...?subscriptionsByQuery,
+        ...?subscriptions[TModel],
+        ...?supabaseControllers,
+      },
+    );
+  }
+
+  /// Supabase's realtime payload only returns unique columns;
+  /// the instance must be discovered from these values so it
+  /// can be deleted by all providers.
+  @protected
+  @visibleForOverriding
+  @visibleForTesting
+  Query queryFromSupabaseDeletePayload(
+    Map<String, dynamic> payload, {
+    required Map<String, RuntimeSupabaseColumnDefinition> supabaseDefinitions,
+  }) {
+    final columnsToFields = supabaseDefinitions.entries.fold(<String, String>{}, (acc, entry) {
+      acc[entry.value.columnName] = entry.key;
+      return acc;
+    });
+
+    final fieldsWithValues = payload.entries.fold(<String, dynamic>{}, (acc, entry) {
+      if (columnsToFields[entry.key] == null) return acc;
+      acc[columnsToFields[entry.key]!] = entry.value;
+      return acc;
+    });
```

Comment:
optionally; i think this would be easier to read inverted
```suggestion
      if (columnsToFields[entry.key] != null) {
        acc[columnsToFields[entry.key]] = entry.value;
      }
      return acc;
    });
```